lib/improver/metadata/utilities.py

```diff
@@ -30,10 +30,64 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """General IMPROVER metadata utilities"""
 
+import dask
 import hashlib
+import iris
 # Usage of pickle in this module is only for creation of pickles and hashing
 # the contents. There is no loading pickles which would create security risks.
 import pickle  # nosec
+import numpy as np
+
+
+def create_new_diagnostic_cube(
+        name, units, coordinate_template, attributes=None, data=None):
+    """
+    Creates a template for a new diagnostic cube with suitable metadata.
+
+    Args:
+        name (str):
+            Standard or long name for output cube
+        units (str or cf_units.Unit):
+            Units for output cube
+        coordinate_template (iris.cube.Cube):
+            Cube from which to copy dimensional and auxiliary coordinates
+        attributes (dict or None):
+            Dictionary of attribute names and values
+        data (np.ndarray or None):
+            Data array.  If not set, cube is filled with zeros using a lazy
+            data object, as this will be overwritten later by the caller
+            routine.
+
+    Returns:
+        iris.cube.Cube:
+            Cube with correct metadata to accommodate new diagnostic field
+    """
+    if data is None:
+        data = dask.array.zeros_like(
+            coordinate_template, dtype=np.float32, chunks=-1)
```

Comment:
@TomekTrzeciak,
I don't think we necessarily want to change the chunk size here, we probably want to chunk based on the chunking of the incoming data? Similarly, we might not want to force the dtype to be `np.float32`, but given our use case that might be the safest thing to do.
```
data = da.zeros_like(coordinate_template.core_data(), dtype=np.float32)
```
(NB: `chunks='auto'` should be the same as doing `chunks=coordinate_template.core_data().chunks`, but it turns out that it's not... (in the current version of dask in the production stack), although we perhaps we want to be explicit if we decide to inherit this).

(Also, do we want to use `zeros_like`, or `empty_like`?)

Reply:
@arh89, these are good points. We could just do this:
```python
data = da.zeros_like(coordinate_template.core_data())
```
This does pick up `dtype`  and `chunks` (if template data is lazy) from the template cube.

However, I wonder what if the `data` argument become a mandatory one instead. This would force calculating data array before the cube creation, which in turn would push the code towards separating numerical calculations apart from cube metadata stuff. That would be the theory at least. Could easily end up like this:
```python
cube = cube.copy()
some_in_place_routine(cube)
cube = create_new_diagnostic_cube(
    'new_name', cube.units, cube, cube.attributes, cube.data)
```
which would defeat the purpose

Reply:
Agreed on removing the additional args - depends if we'd rather inherit or enforce a float32 dtype. Inheritance seems nicer, but enforcement might cut down on some of the accidental auto-promotion incidents. Let's go for what you suggest and remove the `dtype` as well.

I need to think a little about the second point - agree with what you're saying about pushing towards data/metadata separation, but understand that we could end up with something that doesn't help us.

Reply:
I think there should be some way to specify a different `dtype`. How else to create non-float32 stuff like, e.g., weather symbols?

Reply:
Good point - okay, let's just inherit unless something is passed in when calling the function
ie:
```
create_new_diagnostic_cube(...,
                                              dtype=None):
data = da.zeros_like(template_cube.core_data(), dtype=dtype)
```
(`dtype=None` acts the same as `dtype=template_cube.core_data().dtype`)

Reply:
Perhaps it's better to always require `dtype` argument. That way there is no danger of using wrong type if it has to be passed explicitly and one can still pick it up from the template by passing `dtype=None` where this is truly desired.

Reply:
I could live with that. Follows "explicit is better than implicit"

Reply:
Added datatype argument that defaults to `np.float32`.  Also removed `chunk` from dask call.

Reply:
Thanks @cgsandford